api/swarm.go

```diff
@@ -0,0 +1,16 @@
+package api
+
+import (
+	"net/http"
+	"net/url"
+)
+
+func (a *API) swarmRedirect(w http.ResponseWriter, req *http.Request) {
+	var err error
+	req.URL, err = url.ParseRequestURI(a.dockerURL)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
We will probably want to log this internally with logrus so an admin knows there is a proxy failure.